### PR TITLE
Fix h2 db url liquibase closes #2517

### LINKED ISSUE #2517: NPE executing liquibase:diff if oauth2 is selected as authentication method
### Reproduce
 - create new jhipster project
 - select H2 in disk for dev environment  (not tested on mysql or other dbs)
 - select oauth2 as authentication method
 - `mvn` (to initialize the database)
 - `mvn liquibase:diff`

### Expected
a diff changelog

### Result
NPE during execution

```
[ERROR] Failed to execute goal org.liquibase:liquibase-maven-plugin:3.4.1:diff (default-cli) on project jhipster: Error setting up or running Liquibase: liquibase.command.CommandExecutionException: java.lang.NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.liquibase:liquibase-maven-plugin:3.4.1:diff (default-cli) on project jhipster: Error setting up or running Liquibase: liquibase.command.CommandExecutionException: java.lang.NullPointerException
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error setting up or running Liquibase: liquibase.command.CommandExecutionException: java.lang.NullPointerException
	at org.liquibase.maven.plugins.AbstractLiquibaseMojo.execute(AbstractLiquibaseMojo.java:398)
	at org.liquibase.maven.plugins.LiquibaseDatabaseDiff.execute(LiquibaseDatabaseDiff.java:146)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: liquibase.exception.LiquibaseException: liquibase.command.CommandExecutionException: java.lang.NullPointerException
	at liquibase.integration.commandline.CommandLineUtils.doDiffToChangeLog(CommandLineUtils.java:197)
	at liquibase.integration.commandline.CommandLineUtils.doDiffToChangeLog(CommandLineUtils.java:174)
	at org.liquibase.maven.plugins.LiquibaseDatabaseDiff.performLiquibaseTask(LiquibaseDatabaseDiff.java:182)
	at org.liquibase.maven.plugins.AbstractLiquibaseMojo.execute(AbstractLiquibaseMojo.java:394)
	... 23 more
Caused by: liquibase.command.CommandExecutionException: java.lang.NullPointerException
	at liquibase.command.AbstractCommand.execute(AbstractCommand.java:13)
	at liquibase.integration.commandline.CommandLineUtils.doDiffToChangeLog(CommandLineUtils.java:195)
	... 26 more
Caused by: java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Matcher.java:1283)
	at java.util.regex.Matcher.reset(Matcher.java:309)
	at java.util.regex.Matcher.<init>(Matcher.java:229)
	at java.util.regex.Pattern.matcher(Pattern.java:1093)
	at liquibase.diff.output.StandardObjectChangeFilter$Filter.matches(StandardObjectChangeFilter.java:104)
	at liquibase.diff.output.StandardObjectChangeFilter.include(StandardObjectChangeFilter.java:70)
	at liquibase.diff.output.StandardObjectChangeFilter.includeMissing(StandardObjectChangeFilter.java:51)
	at liquibase.diff.output.changelog.ChangeGeneratorChain.fixMissing(ChangeGeneratorChain.java:28)
	at liquibase.diff.output.changelog.ChangeGeneratorFactory.fixMissing(ChangeGeneratorFactory.java:99)
	at liquibase.diff.output.changelog.DiffToChangeLog.generateChangeSets(DiffToChangeLog.java:140)
	at liquibase.diff.output.changelog.DiffToChangeLog.print(DiffToChangeLog.java:120)
	at liquibase.diff.output.changelog.DiffToChangeLog.print(DiffToChangeLog.java:71)
	at liquibase.diff.output.changelog.DiffToChangeLog.print(DiffToChangeLog.java:59)
	at liquibase.command.DiffToChangeLogCommand.run(DiffToChangeLogCommand.java:61)
	at liquibase.command.AbstractCommand.execute(AbstractCommand.java:8)
	... 27 more
```

### PR DESCRIPTION
**Pull request by Panthro**

### CODE REVIEW

File: app/templates/_liquibase.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ task liquibaseDiffChangelog(dependsOn: compileJava, type: JavaExec) {
   args "--referenceUrl=hibernate:spring:<%=packageName%>.domain?dialect=<% if (devDatabaseType == 'mysql') { %>org.hibernate.dialect.MySQLInnoDBDialect<% } else if (devDatabaseType == 'postgresql') { %>org.hibernate.dialect.PostgreSQL82Dialect<% } else if (devDatabaseType == 'h2Disk') { %>org.hibernate.dialect.H2Dialect<%} %>"
   args "--username=<% if (devDatabaseType == 'mysql') { %>root<% } else if (devDatabaseType == 'postgresql') { %><%= baseName %><% } %>"
   args "--password="
-  args "--url=<% if (devDatabaseType == 'mysql') { %>jdbc:mysql://localhost:3306/<%= baseName %><% } else if (devDatabaseType == 'postgresql') { %>jdbc:postgresql://localhost/<%= baseName %><% } else if (devDatabaseType == 'h2Disk') { %>jdbc:h2:tcp://localhost:9092/./target/h2db/db:jhipster<% } %>"
+  args "--url=<% if (devDatabaseType == 'mysql') { %>jdbc:mysql://localhost:3306/<%= baseName %><% } else if (devDatabaseType == 'postgresql') { %>jdbc:postgresql://localhost/<%= baseName %><% } else if (devDatabaseType == 'h2Disk') { %>jdbc:h2:file:./target/h2db/db/<%= baseName %><% } %>"
```
Comment (pascalgrimaud) [inline app/templates/_liquibase.gradle]: Have a look at https://github.com/jhipster/generator-jhipster/blob/master/app/templates/src/main/resources/config/_application-dev.yml#L39

I think we should use lowercase

File: app/templates/_pom.xml [hunk 2]
```diff
@@ -785,7 +785,7 @@
                     <changeLogFile>src/main/resources/config/liquibase/master.xml</changeLogFile>
                     <diffChangeLogFile>src/main/resources/config/liquibase/changelog/${maven.build.timestamp}_changelog.xml</diffChangeLogFile>
                     <driver><% if (devDatabaseType == 'mysql') { %>com.mysql.jdbc.Driver<% } else if (devDatabaseType == 'postgresql') { %>org.postgresql.Driver<% } else if (devDatabaseType == 'h2Disk') { %>org.h2.Driver<% } %></driver>
-                    <url><% if (devDatabaseType == 'mysql') { %>jdbc:mysql://localhost:3306/<%= baseName %><% } else if (devDatabaseType == 'postgresql') { %>jdbc:postgresql://localhost/<%= baseName %><% } else if (devDatabaseType == 'h2Disk') { %>jdbc:h2:tcp://localhost:9092/./target/h2db/db/tmp<% } %></url>
+                    <url><% if (devDatabaseType == 'mysql') { %>jdbc:mysql://localhost:3306/<%= baseName %><% } else if (devDatabaseType == 'postgresql') { %>jdbc:postgresql://localhost/<%= baseName %><% } else if (devDatabaseType == 'h2Disk') { %>jdbc:h2:file:./target/h2db/db/<%= baseName %><% } %></url>
```
Comment (pascalgrimaud) [inline app/templates/_pom.xml]: same comment -> lowercase

File: app/templates/_liquibase.gradle (hunk 1 above)
Comment (Panthro) [inline app/templates/_liquibase.gradle]: I srsly had the feeling I copied from that exact line! I'll update the PR

Comment (deepu105) [inline app/templates/_liquibase.gradle]: Dont merge this yet. The url was changed by me earlier to make it work in
linux and windows properly so test in both os as well as mac b4 merging

**Comment by deepu105** [comment]
Ok you are changing only liquibase related +1

**Comment by pascalgrimaud** [comment]
it works! thanks

**Pull request merged**